os/pm/pm_metrics.c

```diff
@@ -17,117 +17,289 @@
  ****************************************************************************/
 
 #include <tinyara/config.h>
+#include <tinyara/pm/pm.h>
 #include <time.h>
 #include <queue.h>
 #include <debug.h>
-#include "pm_metrics.h"
+#include <errno.h>
 #include "pm.h"
 
-time_t time_diff(time_t time1, time_t time2)
+struct pm_metric_domain_s {
+	uint32_t idle_suspend_counts[CONFIG_PM_NDOMAINS];   /* It counts the suspended domain inside idle thread which prevent board sleep */
+	uint32_t total_suspend_counts;                      /* It counts no. of times idle thread was not able to make board sleep */
+	clock_t suspend_ticks[CONFIG_PM_NDOMAINS];          /* Time (in ticks) domain suspended */
+	clock_t dtime[CONFIG_PM_NDOMAINS];                  /* Last suspended time stamp of domain */
+};
+
+typedef struct pm_metric_domain_s pm_metric_domain_t;
+
+struct pm_metric_wakeup_src_s {
+	uint32_t wakeup_src_counts[PM_WAKEUP_SRC_COUNT];    /* It counts the frequency of wakeup sources */
+};
+
+typedef struct pm_metric_wakeup_src_s pm_metric_wakeup_src_t;
+
+struct pm_metric_state_s {
+	clock_t sticks[PM_COUNT];                           /* PM State Time (in ticks)*/
```

Comment:
Please rename 
I just suggest .. occupied ticks?

Reply:
can we make it `state_ticks` ?